ticket_facts.view.lkml

```diff
@@ -0,0 +1,277 @@
+          COUNT(DISTINCT tickets.id) FILTER(WHERE tickets.via__channel = 'api') AS is_auto_created
+
+        FROM
+        zendesk.audits AS audits
+
+        LEFT JOIN zendesk.audits__events AS events
+        ON events.audit_id = audits.id
+
+        LEFT JOIN zendesk.tickets AS tickets
+        ON audits.ticket_id = tickets.id
+
+        LEFT JOIN zendesk.zendesk_ticket_metrics AS metrics
+        ON audits.ticket_id = metrics.ticket_id
+
+        LEFT JOIN zendesk.zendesk_groups AS groups
+        ON tickets.group_id = groups.id
+        GROUP BY 1,2,3,4,5,6,7 ;;
+
+    indexes: ["ticket_id"]
+    sql_trigger_value: SELECT COUNT(*) FROM zendesk.tickets ;;
+   }
+
+    dimension: ticket_id {
+      primary_key: yes
+      type: number
+      hidden: yes
+      sql: ${TABLE}.ticket_id ;;
+    }
+
+    dimension: status {
+      hidden: yes
+      type: string
+      sql: ${TABLE}.status ;;
+    }
+
+    dimension: via__channel {
+      hidden: yes
+      type: string
+      sql: ${TABLE}.via__channel ;;
+    }
+
+    dimension: via__source__rel {
+      hidden: yes
+      type: string
+      sql: ${TABLE}.via__source__rel ;;
+    }
+
+    dimension: ticket_source {
+      description: "How the ticket was created"
+      group_label: "Ticket Details"
+      type: string
+      sql: CASE WHEN ${via__channel} = 'voice' AND ${via__source__rel} = 'inbound' THEN 'Inbound Call'
+           WHEN ${via__channel} = 'voice' AND ${via__source__rel} = 'outbound' THEN 'Outbound Call'
+           WHEN ${via__channel} = 'voice' AND ${via__source__rel} = 'voicemail' THEN 'Inbound Voicemail'
+           WHEN ${via__channel} = 'email' AND ${via__source__rel} IS NULL AND (${requesters.email} = 'no-reply@getaround.com'
+                                    OR ${requesters.email} LIKE '%@shadowfam.com') THEN 'Managed Tickets - Email'
+           WHEN ${via__channel} = 'email' AND ${via__source__rel} IS NULL THEN 'Inbound Email'
+           WHEN ${via__channel} = 'api' AND ${via__source__rel} IS NULL AND ${requesters.email} = 'zendesk@getaround.com' THEN 'Sentinel'
+           WHEN ${via__channel} = 'api' AND ${via__source__rel} IS NULL THEN 'Internally Generated'
+           WHEN ${via__channel} = 'sms' AND ${via__source__rel} IS NULL THEN 'Managed Tickets - SMS'
+           WHEN ${via__channel} = 'mobile_sdk' AND ${via__source__rel} = 'mobile_sdk' THEN 'Inbound Email (from GA App)'
+           WHEN ${via__channel} = 'facebook' AND ${via__source__rel} IN ('message', 'post') THEN 'Facebook Message'
+           WHEN ${via__channel} = 'twitter' AND ${via__source__rel} IN ('direct_message', 'mention') THEN 'Twitter Message'
+           WHEN ${via__channel} = 'web' AND ${via__source__rel} = 'follow_up' THEN 'Follow-up Ticket'
+           WHEN ${via__channel} = 'web' AND ${via__source__rel} IS NULL AND ${number_outbound_emails} > 0 THEN 'Outbound Email'
+           ELSE NULL END;;
+    }
+
+    dimension: group_name {
```

Comment:
Why have this dimension here? Seems redundant with directly joining the groups view. Rather, if we wanted to pull this in, we shouldn't duplicate it into a fact view.

A fact would be something more like `first_group_id` which is something we couldn't pull out from just joining the `tickets` view to the `groups` view.

Reply:
See the patterns used in `getaround_renter_fact` for instance where we extract `first_trip_id` and then join that to the `getaround_trip` view as `first_getaround_trip`.

Reply:
This is a hugely helpful dimension to have.  This logic identifies how a ticket first created (Inbound Call, Twitter Message, etc)... something we haven't been able to do previously.  The ticket group identifies which group the ticket is assigned to (happiness, supervisor, lead, etc).